### ISSUE TITLE
Clarity on Q and affine maps in the notation and algebra

### ISSUE THREAD

**Issue by jlperla** [JuliaDiffEq/PDERoadmap#7]
cc: @ChrisRackauckas  @MSeeker1340  Would love your thoughts on this notational stuff for the documents, to make things clearer.

Some results form our meeting to update tin the document.  These should be added into the #3 comments/todo list.
- [x] Globally, I think we need to be clear about algebra that involves linear vs. affine algebra, how the  ⋅ notation is used for matrix/matrix, matrix/vector, and affine/vector transformations, etc. 
   - In particular, should be get rid of **all** uses of `x⋅y`  where x and y are not vectors?  Should we allow it for matrix-matrix products?
   - If I have an affine map, `Q` applied to a vector `x`, how do we denote it?  `Q x`?  `Q  ⋅ x`, `Q(x)`?  etc.
   - From https://en.wikipedia.org/wiki/Affine_transformation the suggested notation is that x :-> Mx + b  is an affine transformation for a linear transformation M and a vector b.
- [x] It may make sense to define the general affine notation that (5) uses in a more concrete place such as the beginning section.  
- [x] What I had written at the top of section 2 is flat-out wrong.  Q is an affine-map, and cannot be represented by a matrix.   I would make sure to state the mapping types.  that is, is it $Q :  R^\bar{K} \to R^{\bar{K}}$?  Is that right, or is it a K in one of those, etc.?
- [x] Remove the comment in section 2 about "The Q matrix can be decomposed...", etc. to make it correct.  Also, move up the "Q is actually in general affine..." comment up to the top of the section and make it more clear.
- [x] Equation (3) and (4) are important to make clear that this is affine algebra 
- [x] Are any of the defined objects here an affine map on a discrete space here outside of the `Q`?  If there are, lets identify them.
- [x] But the only substantial issue is that (26), (27), and (28) do not seem to match the sample code in the prototype, and we need to figure out why.  Also, can remove (29) as $\bar{u}$ is never necessary.  After we figure this out we can adjust all of the others.

**Comment by jlperla** [comment]
On the first point, sounds like the notation Q ⋅ x is perfectly reasonable for an affine map Q, so we should leave it as such.  However, I think we need to be very careful when writing out equations that have affine components in them when we go through the algebra.

**Comment by jlperla** [comment]
@stevenzhangdx @FernandoKuwer  I am going to take ownership of this issue and do chunks of it myself, then hand it back over.  
> A for affine
> L for linear
> b for bias

Summarizing discussions with Chris on the notation, we will make some global (and hopefully final!) changes to this that make sense.
- In general, $A$ refers to an affine (or possibly affine) operator
- In general, $L$ would refer to a linear operator 
- An operator on a continuous space uses the tilde, so it is $\tilde{A}$ which becomes $A$ when discretized.  Similarly $\tilde{L}$ becomes $L$
- The reusable discretized oerpators for the central differences, etc. will be called L_2, L_1^+, 
- Get rid of all \cdot except with vectors.

**Comment by jlperla** [comment]
@stevenzhangdx @FernandoKuwer  I made many of the changes, as well as a move towards $M$ for the index instead of $K$.  I can't remember why we didn't use that before, but it makes plenty of sense (often $N$ is used for the time grid).

See the unfinished task above, which was the most important one.  When you go through the document, I would double-check that I made the notation changes consistently, but I think we are basically there.
